Nos/Service/Relay/RelaySubscriptionManager.swift

```diff
@@ -270,28 +300,48 @@ actor RelaySubscriptionManagerActor: RelaySubscriptionManager {
             return 
         }
         
-        if var priorError = errored[relayAddress] {
+        guard let connection = socketConnections[relayAddress] else {
+            return
+        }
+        
+        if case WebSocketState.errored(var priorError) = connection.state {
             priorError.trackRetry()
-            errored[relayAddress] = priorError
+            connection.state = .errored(priorError)
+            Log.debug("Tracking error on websocket connection to \(relayAddress)")
         } else {
-            errored[relayAddress] = WebsocketErrorEvent()
+            connection.state = .errored(WebSocketErrorEvent())
         }
     }
     
     /// This should be called when a socket is successfully opened. It will reset the error count for the socket
     /// if it was above zero.
-    func markHealthy(socket: WebSocket) {
+    func trackConnected(socket: WebSocket) {
         guard let url = socket.request.url else { 
             return 
         }
         
-        errored.removeValue(forKey: url)
+        guard let connection = socketConnections[url] else {
+            return
+        }
+       
+        let oldState = connection.state
+        guard oldState != .connected else {
+            return
+        }
+        
+        connection.state = .connected
+        Log.debug("\(url) has connected")
+        
+        for subscription in active where subscription.relayAddress == url {
+            requestEvents(from: connection.socket, subscription: subscription)
+        }
+        processSubscriptionQueue()
     }
 }
 
 /// A container that tracks how many times we have tried unsuccessfully to open a websocket and the next time we should
 /// try again.
-fileprivate struct WebsocketErrorEvent {
+struct WebSocketErrorEvent: Equatable {
```

Comment:
I generally prefer smaller files, so I'd love to see this struct and everything below in a separate file. One hidden benefit: smaller files make PR reviews easier since I do them one file at a time and check the "Viewed" box once I'm good with a file.